query/src/main/java/org/infinispan/query/SearchManager.java

```diff
@@ -29,41 +29,25 @@
     */
    <E> CacheQuery<E> getQuery(String queryString);
 
-   /**
-    * The MassIndexer can be used to rebuild the Lucene indexes from the entries stored in Infinispan.
-    *
-    * @return the MassIndexer component
-    */
-   MassIndexer getMassIndexer();
 
    /**
-    * Get access to the Query specific statistics for this SearchManager instance
+    * Provides the Hibernate Search DSL entrypoint to build full text queries.
     *
-    * @return The statistics.
-    * @since 7.0
+    * @param entityType A collection of indexed types, or supertypes of all indexed types that will be targeted by the search query.
+    * @param <E> A supertype of all indexed types that will be targeted by the search query.
+    * @return {@link SearchQuerySelectStep}
     */
-   Statistics getStatistics();
+   <E> SearchQuerySelectStep<?, EntityReference, E, ?, ?, ?> search(Class<E> entityType);
```

Comment:
We will only support Icke queries on 11

Reply:
More precisely, the plan is to remove SearchManager altogether and have no Lucene/Hibernate Search classes as part of the public API